soh/soh/Enhancements/randomizer/hint.cpp

```diff
@@ -28,17 +28,22 @@ HintType Hint::GetHintType() {
     return hintType;
 }
 
-void Hint::SetHintedRegion(Text region) {
-    hintedRegion = region;
+void Hint::SetHintedArea(RandomizerArea area) {
+    hintedArea = area;
 }
 
-std::string Hint::GetHintedRegion() {
-    return hintedRegion.GetEnglish();
+RandomizerArea Hint::GetHintedArea() {
+    return hintedArea;
+}
+
+void Hint::SetDistribution(std::string distributionName) {
+    distribution = distributionName;
+}
+
+std::string Hint::GetDistribution() {
+    return distribution;
```

Comment:
Consider returning a `const std::string&` here instead, so we're passing a reference/pointer around rather than copying this string around.